### DISCUSSION TITLE
Include parameters when accepting suggestion of a function from completions list

### DISCUSSION THREAD

**Discussion by stolenzc** [microsoft/pylance-release#3986] [Enhancement]
I don't know is Pylance support this function? But I can't find how to set this.

If I ban to use pylance or change the language server to jedi, when I choose a function by suggestion, it will autofill parentheses and autofill parameter name when definition function, just like this:
![image](https://user-images.githubusercontent.com/42373706/220385866-5a468330-351a-48af-a760-f80d4982ca5f.png)

but when I use pylance, I don't know how to make them autofill the parameters name, I just can set autofill parentheses by completeFunctionParens option, I think autofill parameter name is a good function, I want to use it, I hope who can help me

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by heejaechang** [comment]
what do you do with the `obj` and `class_or_tuple` that don't exist yet?

**Reply by luabud** [reply]
the idea is that `obj` and `class_or_tuple` are placeholders. You can start typing right away after hitting tab to insert the value for `obj`, and then hit tab again to type the value you want to pass for `class_or_tuple`

![autofillparamswithjedi](https://github.com/microsoft/pylance-release/assets/45497113/3acfc788-3fbf-4e44-a881-5fbcf3feba87)

**Reply by heejaechang** [reply]
ah, it is `snippet`. sounds like a cool feature. we can use `placeholder` feature in LSP - https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionClientCapabilities - to implement it if we want to.
